### DISCUSSION TITLE
Изменение формата даты в множественном селекте

### DISCUSSION THREAD

**Discussion by Vadimgrig2012** [totumonline/totum-ru-issues-and-discussions#349] [Как это сделать?] [answered]
Всем добрый день. Подскажите пожалуйста, как правильно изменить формат даты в множественном селекте? Есть поле селект, которое отображает множественные значения дат, но вот в таком формате: 2023-04-12, ... 
Селект берет значения из поля в текущей таблице, но через bfield. 
Я понимаю, что это нужно делать через dateformar, но вот не совсем могу понять в каком моменте эта функция должна стоять в селекте...
Значение селекта не будет изменяемым и будет отображаться без возможности редактирования. Просто отображение корректного списка дат.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
В dateFormat появилась возможность обрабатывать list и rowList:

Следующие параметры применяются если в date используется list/rowList:

+ **keys** — для rowList, список ключей, в которых будет произведена замена. Если не указан, то замена производится во всех ключах.

+ **recursive** — по умолчанию true или list в котором указаны уровни замены 0 — верхний, 1 — следующий вложенный и так далее, для случаев, когда нужно заменить на определенном уровне волженности.

+ **replace** — может содержать или одинарный элемент или тройственный:

    + `"single-part"` — на это значение будут заменены null и "".

    + `"three-part" = "value"` — в этом случае он сверяет значение и заменяет только при соответствии. Имеет приоритет перед форматированием даты.

```
= : dateFormat(date: $rowlist; format: "d.Y"; keys: json`["format"]`)

rowlist: rowListCreate(field: "format" = $list; field: "out" = $list)
    ~list: listCreate(item: "2023-01-01"; item: "2023-01-02"; item: "2023-01-03")

// Результат:
//[
//  {
//    "out": "2023-01-01",
//    "format": "01.2023"
//  },
//  {
//    "out": "2023-01-02",
//    "format": "02.2023"
//  },
//  {
//    "out": "2023-01-03",
//    "format": "03.2023"
//  }
//]
```
